src/test/scala/org/broadinstitute/dsde/workbench/sam/directory/PostgresDirectoryDAOSpec.scala

```diff
@@ -594,5 +594,100 @@ class PostgresDirectoryDAOSpec extends FreeSpec with Matchers with BeforeAndAfte
         dao.loadUser(defaultUser.id).unsafeRunSync().flatMap(_.googleSubjectId) shouldBe Option(newGoogleSubjectId)
       }
     }
+
+    "loadSubjectFromEmail" - {
+      "load a user subject from their email" in {
+        dao.createUser(defaultUser).unsafeRunSync()
+
+        dao.loadSubjectFromEmail(defaultUser.email).unsafeRunSync() shouldBe Some(defaultUser.id)
+      }
+
+      "load a group subject from its email" in {
+        dao.createGroup(defaultGroup).unsafeRunSync()
+
+        dao.loadSubjectFromEmail(defaultGroup.email).unsafeRunSync() shouldBe Some(defaultGroupName)
+      }
+
+      "load a pet service account subject from its email" in {
+        dao.createUser(defaultUser).unsafeRunSync()
+        dao.createPetServiceAccount(defaultPetSA).unsafeRunSync()
+
+        dao.loadSubjectFromEmail(defaultPetSA.serviceAccount.email).unsafeRunSync() shouldBe Some(defaultPetSA.id)
+      }
+
+      "load a policy subject from its email" is pending
+
+      "throw an exception when an email refers to more than one subject" in {
+        dao.createUser(defaultUser).unsafeRunSync()
+        dao.createPetServiceAccount(defaultPetSA.copy(serviceAccount = defaultPetSA.serviceAccount.copy(email = defaultUser.email))).unsafeRunSync()
+
+        assertThrows[WorkbenchException] {
+          dao.loadSubjectFromEmail(defaultUser.email).unsafeRunSync() shouldBe Some(defaultPetSA.id)
+        }
+      }
+    }
+
+    "loadSubjectFromGoogleSubjectId" - {
+      "load a user subject from their google subject id" in {
+        dao.createUser(defaultUser).unsafeRunSync()
+
+        dao.loadSubjectFromGoogleSubjectId(defaultUser.googleSubjectId.get).unsafeRunSync() shouldBe Some(defaultUser.id)
+      }
+
+      "load a pet service account subject from its google subject id" in {
+        dao.createUser(defaultUser).unsafeRunSync()
+        dao.createPetServiceAccount(defaultPetSA).unsafeRunSync()
+
+        //todo: harmonize types
```

Comment:
do this todo?

Reply:
i'm actually not going to do this. the current Ldap implementation has the same issue